apps/meteor/server/models/raw/LivechatContacts.ts

```diff
@@ -61,11 +110,132 @@ export class LivechatContactsRaw extends BaseRaw<ILivechatContact> implements IL
 		);
 	}
 
+	async findContactMatchingVisitor(visitor: AtLeast<ILivechatVisitor, 'visitorEmails' | 'phone'>): Promise<ILivechatContact | null> {
+		const emails = visitor.visitorEmails?.map(({ address }) => address).filter((email) => Boolean(email)) || [];
+		const phoneNumbers = visitor.phone?.map(({ phoneNumber }) => phoneNumber).filter((phone) => Boolean(phone)) || [];
+
+		if (!emails?.length && !phoneNumbers?.length) {
+			return null;
+		}
+
+		const query = {
+			$and: [
+				{
+					$or: [
+						...emails?.map((email) => ({ 'emails.address': email })),
+						...phoneNumbers?.map((phone) => ({ 'phones.phoneNumber': phone })),
+					],
+				},
+				{
+					$or: [
+						{
+							channels: { $exists: false },
+						},
+						{
+							channels: { $size: 0 },
+						},
+					],
```

Comment:
I wonder if we can have a better control of this field so we don't need to test both conditions and have better index coverage..

like if we can make sure the fiel is always there, this query could be indexed and just this:
```suggestion
					channels: []
```

Reply:
Done!